components/layout/flow.rs

```diff
@@ -76,6 +76,30 @@ use table_wrapper::TableWrapperFlow;
 #[allow(unsafe_code)]
 pub unsafe trait HasBaseFlow {}
 
+/// Methods to get the `BaseFlow` from any `HasBaseFlow` type.
+pub trait GetBaseFlow {
```

Comment:
Why not let `HasBaseFlow` become a super trait of `GetBaseFlow` instead?

Reply:
Making this `trait GetBaseFlow: HasBaseFlow` works, but I'm not sure if it adds anything. There's no technical reason for this trait to require HasBaseFlow for all impls.